.github/actions/install-rust/main.js

```diff
@@ -0,0 +1,28 @@
+// Author: Alex Crichton <alex@alexcrichton.com>
+const child_process = require('child_process');
+const toolchain = process.env.INPUT_TOOLCHAIN;
+
+if (process.platform === 'darwin') {
+  child_process.execSync(`curl https://sh.rustup.rs | sh -s -- -y --default-toolchain=none --profile=minimal`);
+  const bindir = `${process.env.HOME}/.cargo/bin`;
+  console.log(`::add-path::${bindir}`);
+  process.env.PATH = `${process.env.PATH}:${bindir}`;
+}
+
+child_process.execFileSync('rustup', ['set', 'profile', 'minimal']);
+child_process.execFileSync('rustup', ['update', toolchain, '--no-self-update']);
+child_process.execFileSync('rustup', ['default', toolchain]);
+
+// Deny warnings on CI to keep our code warning-free as it lands in-tree. Don't
+// do this on nightly though since there's a fair amount of warning churn there.

```

Comment:
The smart way to do this 😆 docs.rs had so many crates break because of `deny(warnings)` we started passing `--cap-lints allow`.